core/iwasm/fast-jit/fe/jit_emit_control.c

```diff
@@ -595,12 +594,18 @@ handle_next_reachable_block(JitCompContext *cc, uint8 **p_frame_ip)
         }
         else if (block->incoming_insns_for_end_bb) {
             *p_frame_ip = block->wasm_code_end + 1;
-            return handle_op_end(cc, p_frame_ip, false);
+            return handle_op_end(
+                cc, p_frame_ip, block == jit_block_stack_top(&cc->block_stack));
```

Comment:
Not sure why adding `block == jit_block_stack_top(&cc->block_stack)`, for case below:
```wast
(module
  (type $0 (func (result i32)))
  (export "test" (func $1))
  (func $1 (type $0)
    (result i32)
    i32.const 5678
    (block $l (param i32) (result i32)
      i32.const 1234
      i32.add
      i32.const 1
      br 0
      drop
      i32.const 666
    )
  )
)
```
Two 'JMP' instructions are generated in L2:
```
L2:
    ; PREDS(L0)
    ; BEGIN_BCIP=0x0027 END_BCIP=0x0038
    ADD             i7, 0x162e, 0x4d2
    ADD             I21, I0, 0x90L
    STI32           0x1, I21, 0x0
    JMP             L3
    JMP             L3
    ; SUCCS(L3)
```